casper/src/main/scala/coop/rchain/casper/api/BlockApiImpl.scala

```diff
@@ -174,11 +179,63 @@ class BlockApiImpl[F[_]: Concurrent: RuntimeManager: BlockDagStorage: BlockStore
   }
 
-  override def deployStatus(deployId: DeployId): F[ApiErr[String]] =
-    BlockDagStorage[F]
-      .asInstanceOf[BlockDagKeyValueStorage[F]]
-      .deployStatus(deployId)
-      .map(_.asRight[Error])
+  override def deployStatus(deployId: DeployId): F[ApiErr[DeployExecStatus]] = {
+
+    def deployBlockHash: F[String] =
+      findDeploy(deployId).map(_.getOrElse(LightBlockInfo.defaultInstance)).map(_.blockHash)
+
+    def getBlockInfo(blockHash: String): F[BlockInfo] =
+      getBlock(blockHash).map(_.getOrElse(BlockInfo.defaultInstance))
+
+    def getSeqPar(blockHash: String): F[Seq[Par]] =
+      for {
+        par       <- Sync[F].delay(GDeployId(deployId).asInstanceOf[Par])
+        dataAtPar <- getDataAtPar(par, blockHash, usePreStateHash = false)
+        seqPar    <- dataAtPar.leftMap(new Exception(_)).liftTo[F].map { case (seqPar, _) => seqPar }
+      } yield seqPar
+
+    def withBlockInfoAndSeqPar[A](f: (BlockInfo, Seq[Par]) => A) =
+      for {
+        blockHash <- deployBlockHash
+        blockInfo <- getBlockInfo(blockHash)
+        seqPar    <- getSeqPar(blockHash)
+      } yield f(blockInfo, seqPar)
+
+    def notProcessed(message: String) = DeployExecStatus().withNotProcessed(NotProcessed(message))
+
+    for {
+      status      <- BlockDagStorage[F].asInstanceOf[BlockDagKeyValueStorage[F]].deployStatus(deployId)
+      deployIdStr = deployId.toHexString
+      response <- status match {
+                   case "Finalized" =>
+                     withBlockInfoAndSeqPar {
+                       case (blockInfo, seqPar) =>
+                         val success = ProcessedWithSuccess(seqPar, blockInfo, finalized = true)
+                         DeployExecStatus().withProcessedWithSuccess(success)
+                     }
+                   case "Orphaned" =>
+                     withBlockInfoAndSeqPar {
+                       case (blockInfo, _) =>
+                         val message = s"""Deploy $deployIdStr is published in invalid block."""
+                         val error   = ProcessedWithError(message, blockInfo, orphaned = true)
+                         DeployExecStatus().withProcessedWithError(error)
+
+                     }
+                   case "Included" =>
+                     withBlockInfoAndSeqPar {
+                       case (blockInfo, _) =>
+                         val message =
+                           s"""Deploy $deployIdStr was stored in deploy-index but not finalized."""
+                         val error = ProcessedWithError(message, blockInfo)
+                         DeployExecStatus().withProcessedWithError(error)
+                     }
+                   case "Pooled" =>
+                     notProcessed(s"""Deploy "$deployIdStr" stored but not processed.""").pure
+                   case "Expired"     => notProcessed(s"""Deploy "$deployIdStr" was expired.""").pure
+                   case "Unknown" | _ => notProcessed(s"""Deploy "$deployIdStr" not found.""").pure
```

Comment:
Don't wrap enumerated string. This status will be used in client code to detect status so it must be simple constant not the sentence that needs to be parsed.